Formula/vsh.rb

```diff
@@ -0,0 +1,19 @@
+class Vsh < Formula
+  desc "HashiCorp Vault interactive shell"
+  homepage "https://github.com/fishi0x01/vsh"
+  url "https://github.com/fishi0x01/vsh/archive/v0.7.2.tar.gz"
+  sha256 "f20885425a9c6ee04ed447046e6df113585469b37332e02b7a23639b718314a1"
+  license "MIT"
+
+  depends_on "go" => [:build]
+
+  def install
+    system "go", "build", "-ldflags", "-X main.vshVersion=v#{version}", "-o", "build/vsh"
+    bin.install "build/vsh"
+  end
+
+  test do
```

Comment:
Can you propose a more substantial test than checking the version or calling `-h`? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
Thank you for the fast feedback - highly appreciated! :bow: 

`vsh` relies on a heavy [integration test suite](https://github.com/fishi0x01/vsh#quality) which requires [bats](https://github.com/bats-core/bats-core) and docker to run. `vsh` does not work without a Hashicorp vault backend. The `vsh` [Makefile](https://github.com/fishi0x01/vsh/blob/master/Makefile) supports a `make get-bats` and `make integration-tests` command, which respectively download bats and start a vault container to run the integration tests against. However, that vault container also uses a local port, i.e., the tests will fail if that port is already in use. Further, the integration tests take like [10 minutes](https://github.com/fishi0x01/vsh/runs/1713416017?check_suite_focus=true) to run, because they check compatibility for multiple vault versions. 

All in all, I figured the integration tests have a too heavy footprint to include them here. Truth be told, I am not sure how I can tackle this.

Reply:
We don't need a full suite of unit tests in the test block. It just needs to be something that attempts to access vsh's basic functionality.

If you need a Hashicorp vault backend, then you can do a `depends_on "vault" => :test` to make one available in the test block. (It might even make more sense to just declare that dependency unconditionally if there's no point in installing vsh if you don't also have vault.) You could then try to run a simple command/script that makes use of that backend.

Alternatively, you can try to exercise some functionality without the vault backend, and write the test to check that it gives you the error message you expect.

Reply:
Thank you for the explanation!

Vault is usually a remote server, i.e., not local on your machine, so I wouldn't put it as a mandatory local dependency. 

I am trying to launch a vault backend locally and run vsh against it, like you suggest:

```rb
system "vault", "server", "-dev", "&"
system "#{bin}/vsh -c some-command"
```

However, the `&` seems to be neglected and the process is not put to background. 
I couldn't find anything related to parallelization in tests in the documentation. Is there a recommended way to put a process to the background so I can start the vault server and run a test against it?

Reply:
This should be possible, but I'm not very familiar with the details. I think you can use either:
1. `IO.popen` and `expect` -- see `mpd` or `minidlna` for examples; or,
2. `fork` -- `grep fork` in the core repo should give many examples of this.

I think the first option is preferred, but @gromgit will know this better if they have time to chime in.

Reply:
The test is looking promising, but I realised there is one drawback to using `vault` as a test dependency: that would prevent the bottling of this for ARM, which currently doesn't have a bottle (pre-compiled binary package) yet.

Maybe we could try this instead?

>Alternatively, you can try to exercise some functionality without the vault backend, and write the test to check that it gives you the error message you expect.

Or, we could just do the `vault` tests on Intel and skip them on ARM.

Reply:
Ok I see. Thank you so much for your feedback and advice! You helped me out a lot :bow: 

I commented out the working `ls` test, so once vault is bottled, it could be used again.